tests/eventually.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Utility function to retry an assertion until it succeeds or times out.
+ * @param callback - The assertion callback that might throw an AssertionError
+ * @param message - Optional custom error message
+ * @param timeout - Optional timeout in milliseconds (default: 5000ms)
+ */
+export async function eventually<T>(
+  callback: () => T | Promise<T>,
+  message?: string,
+  timeout_ms: number = 5000,
+  delay_ms: number = 15,
+): Promise<T> {
+  const startTime = Date.now();
+
+  while (true) {
+    try {
+      return await callback();
```

Comment:
What if it takes more than timeout_ms to complete? Or, like it never ever resolves? 

Look into Promise.race() to race callback() and setTimeout((resolve), timeout_ms).

Reply:
Thanks, I'll try refactoring in a follow up.